### PR TITLE
[support only 1.21+]update PDB to use policy/v1 as v1beta1 is deprecated

### PR DESCRIPTION
**Pull request by pacoxu**
## Description

https://github.com/kubernetes/kubernetes/pull/99290
The PodDisruptionBudget API has been promoted to policy/v1 with no schema changes. The only functional change is that an empty selector (`{}`) written to a policy/v1 PodDisruptionBudget now selects all pods in the namespace. The behavior of the policy/v1beta1 API remains unchanged. The policy/v1beta1 PodDisruptionBudget API is deprecated and will no longer be served in 1.25+.

## Related issues/PRs

I got the warning message below in `tigera operator` pod

> W0913 09:03:29.509844       1 warnings.go:67] policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+, unavailable in v1.25+; use policy/v1 PodDisruptionBudget

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Update PodDisruptionBudgets to policy/v1 API
```

### CODE REVIEW

File: glide.yaml
Comment (caseydavenport) [inline glide.yaml:10]: Oh weird, I wonder why we still have glide.yaml hanging around!

**Review by caseydavenport — APPROVED** [review]

**Comment by caseydavenport** [comment]
Thanks for the PR!

**Comment by pacoxu** [comment]
PodDisruptionBudget
- policy/v1: is only supported in v1.21+
- policy/v1beta1: is deprecated in 1.21+
- policy/v1beta1: will be removed in 1.25+

If we merge this PR, we only support 1.21+ kubernetes. So we should confirm before merging it.

**Comment by pacoxu** [comment]
A solution for helm is using https://helm.sh/docs/topics/charts/#the-kubeversion-field.

For helm chart
- Before we supports 1.25+ kubernetes, we can set `kubeversion="< 1.25"`.
- After kubernetes 1.25 released, we can set `kubeversion=">= 1.21"`.

**Comment by caseydavenport** [comment]
@pacoxu thanks for the info - yes, I think it's too early to merge this just this minute then. We probably want to merge this PR when v1.23 comes out, since then we'll support v1.21, v1.22, and v1.23.

**Comment by caseydavenport** [comment]
@pacoxu I believe this PR will need to be rebased on the latest monorepo work in this repository, so I'm going to close this for now. Would still love to see this merged, but the relevant files have been moved around.

**Pull request closed without merging**
